packages/@aws-cdk/aws-events/lib/event-pattern.ts

```diff
@@ -33,15 +33,15 @@ export interface EventPattern {
    *
    * @default - No filtering on version
    */
-  readonly version?: string[];
+  readonly version?: any[];
```

Comment:
The big question is whether this is a backward compatible change across all CDK supported languages?

Reply:
I don't think it will be. We probably shouldn't be doing this.

A solution is to use Tokens to force the output of the matcher functions to strings.